mitosheet/mitosheet/code_chunks/step_performers/column_steps/delete_column_code_chunk.py

```diff
@@ -65,4 +74,101 @@ def combine_right(self, other_code_chunk: CodeChunk) -> Optional[CodeChunk]:
+        column_ids = self.get_param('column_ids')
+
+        if reordered_column_id in column_ids:
+            return DeleteColumnsCodeChunk(
+                other_code_chunk.prev_state,
+                self.post_state,
+                self.params,
+                self.execution_data
+            )
+
+        return None
+
+    def _combine_left_add_column_code_chunk(self, other_code_chunk: "AddColumnCodeChunk") -> Optional["CodeChunk"]:
+        if not self.params_match(other_code_chunk, ['sheet_index']):
+            return None
+
+        added_column_id = other_code_chunk.get_param('column_id')
+        column_ids: List[ColumnID] = self.get_param('column_ids')
+
+        if added_column_id in column_ids:
+
+            # Remove the added column, as we can just skip and no longer need to delete
+            new_column_ids = copy(column_ids)
+            column_ids.remove(added_column_id)
+
+            # If there's nothing new, then we return a noop
+            if len(new_column_ids) == 0:
+                return NoOpCodeChunk(
+                    other_code_chunk.prev_state,
+                    self.post_state,
+                    {},
+                    {}
+                )
+            else:
+                # Otherwise, just delete what else is deleted in this step
+                new_params = {
+                    **self.params,
+                    'column_ids': new_column_ids
+                }
+
+                return DeleteColumnsCodeChunk(
+                    other_code_chunk.prev_state,
+                    self.post_state,
+                    new_params,
+                    self.execution_data
+                )
+
+        return None
+
+    def _combine_left_rename_columns_code_chunk(self, other_code_chunk: "RenameColumnsCodeChunk") -> Optional["CodeChunk"]:
+        if not self.params_match(other_code_chunk, ['sheet_index']):
+            return None
+        
+        column_ids_to_new_column_headers: Dict[ColumnID, ColumnHeader] = other_code_chunk.get_param('column_ids_to_new_column_headers')
+        deleted_column_ids: List[ColumnID] = self.get_param('column_ids')
+
+        to_remove_from_rename = []
+        for column_id in deleted_column_ids:
+            if column_id in column_ids_to_new_column_headers:
+                to_remove_from_rename.append(column_id)
+
+        # If there is not any overlap between renamed and deleted, we change nothing
+        if len(to_remove_from_rename) == 0:
+            return None
+
+        # If we are removing all of the renames, then we can just skip the renames, and 
+        # just do the deletes
+        if len(to_remove_from_rename) == len(column_ids_to_new_column_headers):    
+            return DeleteColumnsCodeChunk(
+                other_code_chunk.prev_state,
+                self.post_state,
+                self.params,
+                self.execution_data
+            )
+        
+        # Otherwise, we don't optimize
+        return None
```

Comment:
I wonder in practice how helpful this optimization is going to be without filling this in. But we certainly cover the case where we can do the most optimization, so that's nice